main.go

```diff
@@ -3,66 +3,166 @@ package main
 *     File Name           :     main.go
 *     Created By          :     jonesax
 *     Creation Date       :     [2017-09-26 18:35]
 **********************************************************************************/
+var t *string
+var vars *string
+var output *string
 
 func main() {
 
-	var t = flag.String("template", "", "path to template to populate")
-	var vars = flag.String("varpath", "", "path to var yaml to populate")
-	var output = flag.String("output", "", "name of output file")
+	t := flag.String("template", "", "path to template to populate")
+	vars := flag.String("varpath", "", "path to var yaml to populate")
+	output := flag.String("output", "", "name of output file")
 	flag.Parse()
 
 	if *t == "" || *vars == "" || *output == "" {
 		fmt.Println("vortex is a simple program to combine a template with a yaml file of defined varibles it uses golang {{.var}} format with standard yaml")
 		flag.Usage()
 		return
 	}
-	//Parse template -------------------------------------
-	tout, err := template.ParseFiles(*t)
+
+	if err := InputParametersCheck(t, output, vars); err != nil {
+		log.Fatal(err)
+	}
+
+	if isDirectoryOfTemplates(*t) {
+		if err := ParseDirectoryTemplates(*t, *output, *vars); err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		if err := ParseSingleTemplate(*t, *output, *vars); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+}
+
+func ParseSingleTemplate(tempName string, out string, vars string) error {
+	tout, err := template.ParseFiles(tempName)
 	if err != nil {
-		log.Print(err)
-		return
+		return err
 	}
-	//Create output file ---------------------------------
-	f, err := os.Create(*output)
+
+	outputFileName := filepath.Base(tempName)
+
+	if err := os.Chdir(out); err != nil {
+		return err
+	}
+
+	outputFile, err := os.Create(outputFileName)
 	if err != nil {
-		log.Println("create file: ", err)
-		return
+		return err
 	}
-	defer f.Close()
+	defer outputFile.Close()
 
-	//Read YAML ------------------------------------------
-	bytes, err := ioutil.ReadFile(*vars)
+	workDir := os.Getenv("PWD")
+	os.Chdir(workDir)
+
+	bytes, err := ioutil.ReadFile(vars)
 	if err != nil {
-		log.Print(err)
-		return
+		return err
 	}
+
 	m := make(map[string]interface{})
 	err = yaml.Unmarshal(bytes, m)
 	if err != nil {
-		log.Print("yaml: ", err)
-		return
+		return err
 	}
-	//Execute template ----------------------------------
-	err = tout.Execute(f, m)
+
+	err = tout.Execute(outputFile, m)
 	if err != nil {
-		log.Print("execute: ", err)
-		return
+		return err
 	}
 
 	color.Green("Done")
+
+	return nil
+}
+
+func ParseDirectoryTemplates(tempDirectory string, outDirectory string, vars string) error {
+	files, err := ioutil.ReadDir(tempDirectory)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+		if !f.IsDir() {
+			if err := ParseSingleTemplate(fmt.Sprint(tempDirectory, "/", f.Name()), outDirectory, vars); err != nil {
+				return err
+			}
+		}
```

Comment:
I think this is a missed opportunity to support recursion

You could call `ParseDirectoryTemplates` here